### PR TITLE
Fix for when account not found during sync

### PR DESCRIPTION
**Pull request by tomneedham**
Fixes https://jenkins.owncloud.org/job/upgrade-testing/CONFIG=8.2.11%3Bdaily-master-qa,DB=sqlite/285/console

### CODE REVIEW

File: lib/private/User/SyncService.php [hunk 1]
```diff
@@ -117,7 +117,7 @@ public function run(UserInterface $backend, \Closure $callback) {
 					$this->syncAccount($a, $backend);
 					$this->mapper->update($a);
 				} catch(DoesNotExistException $ex) {
-					$this->createNewAccount($backendClass, $uid);
+					$a = $this->createNewAccount($backendClass, $uid);
```
Comment (DeepDiver1975) [inline lib/private/User/SyncService.php]: since this was not detected by the regular CI process - I assume this is missing unit tests

**Review by DeepDiver1975 — CHANGES_REQUESTED** [review]

File: lib/private/User/SyncService.php (hunk 1 above)
Comment (phil-davis) [reply lib/private/User/SyncService.php]: The regression looks like it happened here:
https://github.com/owncloud/core/pull/29669/files#diff-37958be4e01e8baeb127a59409e1a593L114

Comment (tomneedham) [reply lib/private/User/SyncService.php]: Yep - looking at tests for this

**Comment by tomneedham** [comment]
Added a test to run through the sync service at least once.

**Comment by individual-it** [comment]
restarted drone, now passed

**Pull request merged**